### PR TITLE
Homologated classes and logic between Angular and React sdk

### LINKED ISSUE #28729: Angular SDK: Homologate both SDK - Angular & React
### Parent Issue

https://github.com/dotCMS/core/issues/28404

### Task

Homologate structure of both SDK

- [ ] Naming
- [ ] Models
- [ ] Folders

### Proposed Objective

Same as Parent Issue

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Proposed Changes
* Homologate "context" to shared info between components in both SDK (Angular - React)
* Now both context have this structure

```
{
components: Components Map,
pageAssets: Response of PageAPI,
isInsideEditor: boolean from @dotcms/client
}

```
And have same naming convention

### CODE REVIEW

File: core-web/libs/sdk/react/src/lib/components/Container/Container.tsx [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 import { getContainersData } from '../../utils/utils';
-import { PageProviderContext } from '../PageProvider/PageProvider';
 
 function NoContent({ contentType }: { readonly contentType: string }) {
     return <div data-testid="no-component">No Component for {contentType}</div>;
 }
 
 export interface ContainerProps {
-    readonly containerRef: PageProviderContext['layout']['body']['rows'][0]['columns'][0]['containers'][0];
+    readonly containerRef: DotCMSPageContext['pageAsset']['layout']['body']['rows'][0]['columns'][0]['containers'][0];
```
Comment (zJaaal) [inline core-web/libs/sdk/react/src/lib/components/Container/Container.tsx:12]: Why dont we have a type for the PageAsset?

File: core-web/libs/sdk/react/src/lib/components/Row/Row.spec.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ jest.mock('../Column/Column', () => {
 });
 
 describe('Row', () => {
-    const mockRowData: PageProviderContext['layout']['body']['rows'][0] = {
+    const mockRowData: DotCMSPageContext['layout']['body']['rows'][0] = {
```
Comment (zJaaal) [inline core-web/libs/sdk/react/src/lib/components/Row/Row.spec.tsx:20]: There should be a type for this

File: core-web/libs/sdk/react/src/lib/components/Row/Row.tsx [hunk 3]
```diff
@@ -17,10 +17,10 @@ export interface RowProps {
     /**
      * Row data
      *
-     * @type {PageProviderContext['layout']['body']['rows'][0]}
+     * @type {DotCMSPageContext['layout']['body']['rows'][0]}
      * @memberof RowProps
      */
-    row: PageProviderContext['layout']['body']['rows'][0];
+    row: DotCMSPageContext['pageAsset']['layout']['body']['rows'][0];
```
Comment (zJaaal) [inline core-web/libs/sdk/react/src/lib/components/Row/Row.tsx:23]: same here

File: core-web/libs/sdk/react/src/lib/models/index.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            [key: string]: {
+                contentType: string;
+                identifier: string;
+                title: string;
+                inode: string;
+                onNumberOfPages: number;
+                widgetTitle?: string;
+                baseType: string;
+            }[];
+        };
```
Comment (zJaaal) [inline core-web/libs/sdk/react/src/lib/models/index.ts:22]: do we need types for this?

**Review by zJaaal — APPROVED** [review]
I marked some types that we should make for the types lib

File: core-web/libs/sdk/react/src/lib/components/Container/Container.tsx (hunk 1 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/react/src/lib/components/Container/Container.tsx:12]: We have all these types in @dotcms/angular

But until we have the type library, we cannot "duplicate" them or we will have SonarQ issues

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/sdk/angular/src/lib/models/dotcms.model.ts [hunk 5]
```diff
@@ -82,27 +121,21 @@ export interface DotCMSContentlet {
...
-export interface DotCMSPageAssetContainer {
-    [key: string]: {
-        container: DotCMSContainer;
-        containerStructures: DotCMSContainerStructure[];
-        contentlets: {
-            [key: string]: DotCMSContentlet[];
-        };
-    };
+export interface DotcmsNavigationItem {
+    code?: any;
```
Comment (nicobytes) [inline core-web/libs/sdk/angular/src/lib/models/dotcms.model.ts:125]: @KevinDavilaDotCMS Why do we use any here?

**Review by nicobytes — APPROVED** [review]

**Pull request merged**